### PR TITLE
Fixed debt, category, transaction forms and list view.

### PR DESCRIPTION
**Pull request by ihorko95**
Views of debt (list, create, update), category (list, create, update), transaction (list, create, update) changed.

### CODE REVIEW

**Comment by ihorko95** [comment]
Is it ok?

File: src/core/forms.py [hunk 1]
```diff
@@ -12,15 +12,50 @@ class DebtForm(UserHiddenInput):
...
+        # username = forms.CharField(label="User Name:*", widget=forms.TextInput(attrs={'class': 'form-control'}))
+        # email = forms.EmailField(required=False, label="Email:",
+        #                          widget=forms.EmailInput(attrs={'class': 'form-control', 'requered': 'false'}))
+        # first_name = forms.CharField(required=False, label="First Name:",
+        #                              widget=forms.TextInput(attrs={'class': 'form-control'}))
+        # last_name = forms.CharField(required=False, label="Last Name:",
+        #                             widget=forms.TextInput(attrs={'class': 'form-control'}))
+        # password1 = forms.CharField(label="Password:*", widget=forms.PasswordInput(attrs={'class': 'form-control'}))
+        # password2 = forms.CharField(label="Confirm password:*",
+        #                             widget=forms.PasswordInput(attrs={'class': 'form-control'}))
```
Comment (IvanYukish) [inline src/core/forms.py]: For these things we use, Django widget-tweaks (https://pypi.org/project/django-widget-tweaks/). Please update it and let me know when it'll be done.

File: src/core/forms.py [hunk 2]
```diff
@@ -12,15 +12,50 @@ class DebtForm(UserHiddenInput):
...
         model = Category
-        fields = ['user', 'name', 'description', 'type']
+        fields = ['user', 'name', 'type', 'description']
+        widgets = {
+            'name': forms.TextInput(
+                attrs={'class': 'form-control form-input-height', 'placeholder': 'Category name', 'required': 'True'}),
+            'type': forms.Select(attrs={'class': 'form-control form-input-height', 'placeholder': 'Type'}),
+            'description': forms.Textarea(
+                attrs={'class': 'form-control form-input-height', 'placeholder': 'Description'}),
+        }
```
Comment (IvanYukish) [inline src/core/forms.py]: the same as in the first comment. You need use Django widget tweaks

File: src/core/forms.py [hunk 3]
```diff
@@ -12,15 +12,50 @@ class DebtForm(UserHiddenInput):
...
     class Meta:
         model = Transaction
-        fields = ['prise', 'description', 'category']
+        fields = ['prise', 'category', 'description']
+        widgets = {
+            'prise': forms.NumberInput(attrs={'class': 'form-control form-input-height', 'placeholder': 'Prise'}),
+            'category': forms.Select(attrs={'class': 'form-control form-input-height'}, ),
+            'description': forms.Textarea(
+                attrs={'class': 'form-control form-input-height', 'placeholder': 'Description'}),
+        }
```
Comment (IvanYukish) [inline src/core/forms.py]: the same as in the first comment. You need use Django widget tweaks

File: src/core/views.py [hunk 4]
```diff
@@ -155,6 +155,7 @@ def get_queryset(self):
         return self.model.objects.filter(category__user=self.request.user)
 
 
+
```
Comment (IvanYukish) [inline src/core/views.py]: unused row

File: src/templates/core/category_create.html [hunk 5]
```diff
@@ -1,44 +1,29 @@
...
+{% block card %}
+    <div class="col-lg-12 d-flex justify-content-center mt-3">
+        <h1>{% trans "Add Category" %}</h1>
+    </div>
+ <form action="{% url 'core:category-create' %}" method="post" class="card-body" id="profileUpdateForm">
+        {% csrf_token %}
+        {% for field in form %}
+            <div class="row mb-3 justify-content-center">
+                <div class="col-sm-9 text-secondary">
+                    {{ field }}
```
Comment (IvanYukish) [inline src/templates/core/category_create.html]: Instead iteration thorough form, you should use {{form.field_name }} or {% render_field form.field_name ...another params %}

File: src/templates/core/category_list.html [hunk 6]
```diff
@@ -1,44 +1,55 @@
-{% extends "base/base.html" %}
+{% extends "base/base_card.html" %}
```
Comment (IvanYukish) [inline src/templates/core/category_list.html:1]: I'm not sure that we should use another base block for these forms, as I remember,  "main" block already has a base structure for nested form, anyway, we should discuss it.

File: src/templates/core/category_update.html [hunk 7]
```diff
@@ -1,17 +1,47 @@
...
+{##}
+{#    <h1>{% trans "Update category" %}</h1>#}
+{##}
+{#    <form action="{% url "core:category-update" category.id %}" method="post">#}
+{#        {% csrf_token %}#}
+{#        {{ form.as_p }}#}
+{#        <button type="submit">{% trans "Update" %}</button>#}
+{#    </form>#}
+{##}
+{#{% endblock %}#}
```
Comment (IvanYukish) [inline src/templates/core/category_update.html]: you should remove it or add #TODO row and explain the reasons why you left it here.

File: src/templates/core/transaction_create.html [hunk 8]
```diff
@@ -1,21 +1,53 @@
...
+{#    <form method="POST" action="{% url 'core:transaction-create' %}">#}
+{#        {% csrf_token %}#}
+{#        {{ form.as_p }}#}
+{#        {% if redirect_field_value %}#}
+{#            <input type="hidden" name="{{ redirect_field_name }}" value="{{ redirect_field_value }}"/>#}
+{#        {% endif %}#}
+{#        <button class="primaryAction" type="submit">{% trans "Add transaction" %}</button>#}
+{#    </form>#}
+{##}
+{#{% endblock %}#}
```
Comment (IvanYukish) [inline src/templates/core/transaction_create.html]: TODO or remove it

File: src/templates/core/transaction_list.html [hunk 9]
```diff
@@ -1,38 +1,86 @@
...
+{#        </div>#}
+{#    {% endfor %}#}
+{##}
+{#    <br>#}
+{##}
+{#    <a type="button" class="btn btn-success btn-sm ml-3"#}
+{#       href="{% url 'core:transaction-create' %}">{% trans "Create new Transaction" %}</a>#}
+{#    {{ data }}#}
+{##}
+{#{% endblock %}#}
```
Comment (IvanYukish) [inline src/templates/core/transaction_list.html]: TODO or remove it

File: src/templates/core/category_list.html (hunk 6 above)
Comment (ihorko95) [reply src/templates/core/category_list.html:1]: If we would like use white card behind, we need use new extends. There is the difference.
Screenshot from 2021-08-16 12-47-17

File: src/core/forms.py (hunk 1 above)
Comment (ihorko95) [reply src/core/forms.py]: I've done it.

File: src/core/forms.py (hunk 2 above)
Comment (ihorko95) [reply src/core/forms.py]: done

File: src/core/forms.py (hunk 3 above)
Comment (ihorko95) [reply src/core/forms.py]: done

File: src/templates/core/category_create.html (hunk 5 above)
Comment (ihorko95) [reply src/templates/core/category_create.html]: done

File: src/templates/core/category_update.html (hunk 7 above)
Comment (ihorko95) [reply src/templates/core/category_update.html]: removed

File: src/templates/core/transaction_create.html (hunk 8 above)
Comment (ihorko95) [reply src/templates/core/transaction_create.html]: removed

File: src/templates/core/transaction_list.html (hunk 9 above)
Comment (ihorko95) [reply src/templates/core/transaction_list.html]: removed

**Pull request merged**